source/slang/slang-emit-c-like.cpp

```diff
@@ -3937,6 +3943,8 @@ void CLikeSourceEmitter::emitFuncDecl(IRFunc* func, const String& name)
     auto funcType = func->getDataType();
     auto resultType = func->getResultType();
 
+    emitFunctionPreambleImpl(func);
+
```

Comment:
Not sure about this change, but it seemed to be missing, as otherwise `__global__` for CUDA kernels was missing in the declaration.